### PR TITLE
Added error callback for propagating generic errors from librdkafka to app

### PR DESCRIPTION
**Pull request by edenhill**

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -78,6 +78,9 @@ The Python bindings also provide some additional configuration properties:
 * ``default.topic.config``: value is a dict of topic-level configuration
   properties that are applied to all used topics for the instance.
 
+* ``error_cb``: Callback for generic/global error events. This callback is served by
```
Comment (ewencp) [inline docs/index.rst]: Should we be including the argument lists for these callbacks?

**Comment by ewencp** [comment]
@edenhill LGTM, but it has very limited testing, just that the setup works. For example, we don't actually evaluate that the expected set of arguments matches. Anything we can do after creating the producer/consumer in the unit tests to evaluate that invoking the callback actually works?

Also, should we start tracking changelogs like we do for other projects (e.g. http://docs.confluent.io/3.0.0/schema-registry/docs/changelog.html). We normally update these just before release since its easier than updating with each patch, but given that clients need to release independently and we're doing a new tag + release for this asap....

**Comment by edenhill** [comment]
Updated with unit test and callback arg documentation.

File: confluent_kafka/src/Consumer.c [hunk 2]
```diff
@@ -55,12 +58,15 @@ static void Consumer_dealloc (Consumer *self) {
...
-	if (self->on_assign)
-		Py_VISIT(self->on_assign);
-	if (self->on_revoke)
-		Py_VISIT(self->on_revoke);
+static int Consumer_traverse (Handle *self,
+			      visitproc visit, void *arg) {
+	if (self->u.Consumer.on_assign)
+		Py_VISIT(self->u.Consumer.on_assign);
+	if (self->u.Consumer.on_revoke)
+		Py_VISIT(self->u.Consumer.on_revoke);
```
Comment (ewencp) [inline confluent_kafka/src/Consumer.c]: Are we missing visiting `on_commit`?

**Comment by ewencp** [comment]
@edenhill One last question, but LGTM otherwise.

**Comment by paulcavallaro** [comment]
Wanted to ping on this pull request because isssue #14 is blocking our adoption of this library. Thanks,

-Paul

**Comment by ewencp** [comment]
@paulcavallaro Should see some follow up soon, sorry for the delay. My last comment was right before @edenhill went on vacation, but he is returning imminently!

**Comment by ewencp** [comment]
@edenhill I'm seeing this:

```
$ examples/integration_test.py localhost:9092
Using confluent_kafka module version 0.9.1 (0x90100)
Using librdkafka version 0.9.1 (0x901ff)
('==============================', 'Verifying Producer', '==============================')
Traceback (most recent call last):
  File "examples/integration_test.py", line 366, in <module>
    verify_producer()
  File "examples/integration_test.py", line 86, in verify_producer
    p = confluent_kafka.Producer(**conf)
cimpl.KafkaException: KafkaError{code=_INVALID_ARG,val=-186,str="Property "error_cb" must be set through dedicated .._set_..() function"}
```

**Comment by ewencp** [comment]
@edenhill Ok, nm, must have had something stale somewhere. Reran in a virtualenv and its fine.

**Comment by edenhill** [comment]
That looks like you used an old confluent_kafka python module version without error_cb support (thus the fallthru to librdkafka)

**Pull request merged**